File: problems/collection/2709.py
# 2709 - PRIME NUMBERS/FIX PYTHON ERROR WHILE ENCONDING (# -*- coding: utf-8 -*- DOESN'T WORKED)
def is_prime(x):
    if x <= 1:
        return False
    for i in range(2, x): # 2 doesn't entry the loop
        if x % i == 0:
            return False
    return True


def main():
    import sys

    sys.stdout = open(sys.stdout.buffer.fileno(), 'w', encoding='utf8')

    win = "You’re a coastal aircraft, Robbie, a large silver aircraft.\n"
    lose = "Bad boy! I’ll hit you.\n"

    while True:
        try:
            V = 0
            M = int(input())
            coins = [int(input()) for _ in range(M)]
            jump = int(input())
            for i in range(M - 1, -1, -jump):
                V += coins[i]
            if is_prime(V):
                sys.stdout.buffer.write(win.encode())
            else:
                sys.stdout.buffer.write(lose.encode())
        except EOFError:
            break

if __name__ == '__main__':
    main()

File: problems/c/2709.c
#include <stdio.h>
#include <locale.h>

int eh_primo(int);

int main(void)
{
    // setlocale(LC_ALL, "Portuguese");
    int i, v, m, pulo, moedas[20];
    char vitoria[] = "You’re a coastal aircraft, Robbie, a large silver aircraft.";
    char derrota[] = "Bad boy! I’ll hit you.";

    while (scanf("%d", &m) != EOF)
    {
        v = 0;
        for (i = 0; i < m; i++)
        {
            scanf("%d", &moedas[i]);
        }

        scanf("%d", &pulo);

        for (i = m - 1; i >= 0; i -= pulo)
        {
            v += moedas[i];
        }

        if (eh_primo(v))
        {
            puts(vitoria);
        }
        else
        {
            puts(derrota);
        }
    }

    return 0;
}

int eh_primo(int n) {
	if (n <= 1) return 0;
    else if (n <= 3) return 1;
    else if (n % 2 == 0 || n % 3 == 0) return 0;
    int i = 5;
    while (i * i <= n) {
		if (n % i == 0 || n % (i + 2) == 0) return 0;
        i += 6;
	}
    return 1;
}
